packages/grafana-ui/src/components/Logs/LogRowContext.tsx

```diff
@@ -13,25 +13,47 @@ import { LogMessageAnsi } from './LogMessageAnsi';
 interface LogRowContextProps {
   row: LogRowModel;
   context: LogRowContextRows;
+  wrapLogMessage: boolean;
   errors?: LogRowContextQueryErrors;
   hasMoreContextRows?: HasMoreContextRows;
   onOutsideClick: () => void;
   onLoadMoreContext: () => void;
 }
 
-const getLogRowContextStyles = (theme: GrafanaTheme) => {
+const getLogRowContextStyles = (theme: GrafanaTheme, wrapLogMessage?: boolean) => {
+  /**
+   * This is workaround for displaying uncropped context when we have unwrapping log messages.
+   * We are using margins to correctly position context. Because non-wrapped logs have always 1 line of log
+   * and 1 line of Show/Hide context switch. Therefore correct position can be reliably achieved by margins.
+   */
+
+  const afterContext = wrapLogMessage
+    ? css`
+        top: -250px;
+      `
+    : css`
+        margin-top: -250px;
+      `;
+
+  const beforeContext = wrapLogMessage
+    ? css`
+        top: 100%;
+      `
+    : css`
+        margin-top: 40px;
+      `;
   return {
     commonStyles: css`
       position: absolute;
-      width: calc(100% + 20px);
-      left: -13px;
       height: 250px;
       z-index: ${theme.zIndex.dropdown};
       overflow: hidden;
       background: ${theme.colors.bg1};
       box-shadow: 0 0 10px ${theme.colors.dropdownShadow};
       border: 1px solid ${theme.colors.bg2};
       border-radius: ${theme.border.radius.md};
+      width: ${wrapLogMessage ? 'calc(100% + 20px)' : 'auto'};
```

Comment:
`commonStyles` are used only in `LogRowContextGroup` but without `wrapLogMessage` being passed (it uses `useStyle()`). Does it mean width/left could be just set here to `auto` or we should pass `wrapLogMessage` when getting these styles?

Reply:
Good catch. It seems that both of these can be removed (and their default value will be set to auto).